Update spruce-example docs for b13.0.0 API changes

alantreadway opened the pull request:
## Summary

Updates the `/spruce-example` command and related guides to reflect API changes made since b12.3.1:

### Key Changes

1. **`itemId` → `itemType`**: Updated all series callback examples (waterfall, range-area, range-bar, OHLC) to use the new `itemType` parameter instead of deprecated `itemId`
2. **`highlighted` → `highlightState`**: Updated all styler examples to use `highlightState` string instead of deprecated `highlighted` boolean
3. **New axis options**: Added documentation for `preferredMin`/`preferredMax` (soft bounds) and grouped-category axis `wrapping`/`truncate` options
4. **Tooltip symbol customization**: Added new section documenting `symbol` option with `marker` and `line` configuration
5. **Module registration**: Added section to examples guide documenting the new `ModuleRegistry.registerModules()` pattern required for all examples
6. **New features**: Added documentation for zoom `onDataChange` strategy, hierarchy highlight states, and series axis binding

### Files Updated

- `tools/prompts/commands/spruce-example/features/recent-features.md`
- `tools/prompts/commands/spruce-example/features/axes.md`
- `tools/prompts/commands/spruce-example/features/tooltips.md`
- `tools/prompts/commands/spruce-example/core-rules.md`
- `tools/prompts/guides/examples.md`

All changes maintain backward compatibility notes where appropriate and include code examples showing the correct modern patterns.

alantreadway committed:
Update example guides.

AG-DavidG reviewed (approved):
LGTM

AG-DavidG commented on tools/prompts/commands/spruce-example/features/recent-features.md:
```diff
@@ -443,6 +443,108 @@ const chart = AgCharts.createFinancialChart({
+
+_Use for_: Treemap and sunburst charts where you want to highlight related nodes in the same hierarchy branch.
+
+### 📊 Series Axis Binding
```
In secondary axes use cases, it is recommended to use readable keys for the axes. Instead of `y` and `y2` these should be `profit` and `revenue` or similar.

alantreadway committed:
Fix PR feedback: use readable axis keys and format files

- Update Series Axis Binding example to use semantic axis names (category, price, volume) instead of generic x, y, y2)
- Run yarn nx format to fix lint/format CI failures
